src/utils.ts

```diff
@@ -76,8 +85,15 @@ export async function getWalletWithProvider({
     throw new Error("unable to create ETH API provider");
   }
 
-  if ((await provider.getNetwork()).chainId !== network.chainId) {
-    throw new Error("Provider / chain mismatch.");
+  let providerChainId: number | undefined;
+  try {
+    providerChainId = (await provider.getNetwork()).chainId;
+  } catch (err) {
+    throw new Error("cannot determine provider chain ID");
```

Comment:
I added this try-catch-throw in hopes users will have an easier time determining why a call failed.